projects/js-packages/script-data/src/utils.ts

```diff
@@ -128,3 +129,13 @@ export function isWpcomPlatformSite() {
 export function isJetpackSelfHostedSite() {
 	return ! NON_SELF_HOSTED_TYPES.includes( getSiteData()?.host );
 }
+
+/**
+ * Check if the current user has a particular permission.
+ *
+ * @param permission - The feature to check.
+ * @return Whether the current user has that permission.
+ */
+export function canUser( permission: keyof CurrentUserData[ 'permissions' ] ): boolean {
```

Comment:
I couldn't get the typechecker to use the JSDoc types, it kept thinking `permission` was `any`, so I added the types in the code. We've got a weird mixture of JSDoc and TS now.

Reply:
A few suggestions here:
- I think calling it `currentUserCan` is better, as that is what it actually is.
- Let us name it `capability` instead of `permission`.
- I would type the param as `keyof CurrentUserData[ 'permissions' ] | ( string & {} )` to allow the auto-completion as well as allow any other strings as caps which we don't know upfront.

<img width="796" alt="image" src="https://github.com/user-attachments/assets/286bf32b-fb49-4b25-981d-686d15e0797e" />

Reply:
Thanks @manzoorwanijk, I was going to ask your thoughts on the type.

> any other strings as caps which we don't know upfront

That's the point though, is that we only want it to be the capabilities we know about, as they'll be the only ones available in the `permissions` object. I suppose other code could add them in? Could they then extend the type?

Reply:
Yeah, if we want to limit it, that is fine as well. The devs will then need to update the type accordingly. Then we can update `CurrentUserData` type to be

```ts
...
permissions: Record< 'manage_options' | 'manage_modules', boolean >;
...
```

Reply:
Great. I've updated that in 983f9f01ae993474901a07e139be3118e2ed00e0